src/fsharp/FSharp.Core/option.fs

```diff
@@ -85,6 +85,10 @@ namespace Microsoft.FSharp.Core
         [<CompiledName("OfNullable")>]
         let ofNullable (value:System.Nullable<'T>) = if value.HasValue then Some value.Value else None
 
+        [<CompiledName("OfString")>]
+        let ofString (value:System.String) =
+            if System.String.IsNullOrWhiteSpace value then None else Some value
```

Comment:
Good question. In my personal experience we always use IsWhitespace, but perhaps that's just for what we've used in the past.

If it's just if null you may as well just use ``OfObj`` - this is designed to be a little more useful than that with regards to dealing with strings specifically.